aggregator/src/main/java/com/redhat/cloud/notifications/DailyEmailAggregationJob.java

```diff
@@ -96,12 +112,8 @@ public void processDailyEmail() {
                 }
             }
```

Comment:
I just noticed how wrong the old code is: we're checking if all `futures` are completed on each loop iteration. It should only be done once after the loop is done iterating.

Reply:
Fixed in https://github.com/RedHatInsights/notifications-backend/pull/2120